amd_openvx_extensions/amd_custom/source/custom_layer.cpp

```diff
@@ -0,0 +1,244 @@
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "kernels.h"
+#include "custom_api.h"
+
+struct CustomLayerLocalData {
+    CustomFunctionType function;
+    customHandle custom_handle;
+    unsigned char *pCustomParameterArray;
+    customTensorDesc input_desc, output_desc;
+    void* input_mem;
+    void* output_mem;
+    unsigned int cpu_num_threads;
+    void * hipstream;
+    customBackend backend;
+};
+
+inline void Set4dTensorDesc(customTensorDesc &desc, int data_type, vx_size (&dims)[4],  vx_size (&strides)[4])
+{
+    desc.data_type = (customDataType)data_type;
+    desc.dims[0] = (unsigned int)dims[0], desc.dims[1] = (unsigned int)dims[1];
+    desc.dims[2] = (unsigned int)dims[2], desc.dims[3] = (unsigned int)dims[3];
+    desc.strides[0] = (unsigned int)strides[0], desc.strides[1] = (unsigned int)strides[1];
+    desc.strides[2] = (unsigned int)strides[2], desc.strides[3] = (unsigned int)strides[3];
+}
+
+static vx_status VX_CALLBACK validateCustomLayer(vx_node node, const vx_reference parameters[], vx_uint32 num, vx_meta_format metas[])
+{
+    // check scalar type
+    vx_enum type, in_type, out_type;
+    ERROR_CHECK_STATUS(vxQueryScalar((vx_scalar)parameters[1], VX_SCALAR_TYPE, &type, sizeof(type)));
+    if (type != VX_TYPE_UINT32) return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: custom: #1 scalar type=%d (not uint32)\n", type);
+    if (parameters[2]) {
+      ERROR_CHECK_STATUS(vxQueryScalar((vx_scalar)parameters[2], VX_SCALAR_TYPE, &type, sizeof(type)));
+      if (type != VX_TYPE_UINT32) return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: custom: #2 scalar type=%d (not uint32)\n", type);
+    }
+    if (parameters[3]) {
+      vx_size itemsize = 0;
+      ERROR_CHECK_STATUS(vxQueryArray((vx_array)parameters[3], VX_ARRAY_ITEMTYPE, &type, sizeof(type)));
+      if(type != VX_TYPE_CHAR) return VX_ERROR_INVALID_TYPE;
+      vx_size capacity;
+      ERROR_CHECK_STATUS(vxQueryArray((vx_array)parameters[3], VX_ARRAY_CAPACITY, &capacity, sizeof(capacity)));
+      if(capacity != 256) return VX_ERROR_INVALID_DIMENSION;
+      ERROR_CHECK_STATUS(vxQueryArray((vx_array)parameters[3], VX_ARRAY_ITEMSIZE, &itemsize, sizeof(itemsize)));
+      if(itemsize != sizeof(VX_TYPE_CHAR)) return VX_ERROR_INVALID_TYPE;
+    }
+
+    // check tensor dimensions
+    vx_size num_dims;
+    vx_size input_dims[4], output_dims[4];
```

Comment:
initialize the arrays by adding ={}

Reply:
I am not sure if this needed since it will be filled by the following API. It is wasting compute cycles.